sqlglot/dialects/mysql.py

```diff
@@ -60,9 +61,20 @@ def _date_trunc_sql(self: MySQL.Generator, expression: exp.DateTrunc) -> str:
     return f"STR_TO_DATE({concat}, '{date_format}')"
 
 
-def _str_to_date(args: t.List) -> exp.StrToDate:
-    date_format = MySQL.format_time(seq_get(args, 1))
-    return exp.StrToDate(this=seq_get(args, 0), format=date_format)
+# All specifiers for time parts (as opposed to date parts)
+# https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_date-format
+TIME_SPECIFIER_PATTERN = re.compile(r"(?<!%)%[fHhIiklprSsT]")
```

Comment:
why do you need negative look back?

Reply:
I removed the regex entirely. But it was a failed attempt at handling an escaped `%%`